### PR TITLE
lib/x11utils.pm: Handle gnome-activities in ensure_unlocked_desktop

### PR DESCRIPTION
**Pull request by GraceWang571**
Handle gnome-activities in ensure_unlocked_desktop
Update the codes of handling gnome-activities in handle_login

Missing in #12883, as I mentioned in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12924#issuecomment-883257409, the multi_users_dm failed caused. ensure_unlocked_desktop need to be updated to handle gnome-activities

- Needles: N/A
- Verification run: 
- extra_tests_gnome https://openqa.opensuse.org/tests/1848768#step/multi_users_dm/47
- otherDE_lxqt https://openqa.opensuse.org/tests/1849887
- extra_tests_on_kde https://openqa.opensuse.org/tests/1849877

### CODE REVIEW

**Comment by punkioudi** [comment]
Hi Grace! Could you also add the VRs for Leap 15.3 `extra_tests_on_kde` ?

**Comment by GraceWang571** [comment]
> Hi Grace! Could you also add the VRs for Leap 15.3 `extra_tests_on_kde` ?

Sure, but I am afraid it may take time. Since almost all the workers are now occupied with running the latest build 20210720 tests.
I prepared to provide VR for Leap 15.3 `extra_tests_on_kde`, but I failed. See https://openqa.opensuse.org/tests/1848769
I just cloned a new one https://openqa.opensuse.org/tests/1849877 (based on https://openqa.opensuse.org/tests/1849351), it's now in scheduled status.

**Comment by DimStar77** [comment]
> Sure, but I am afraid it may take time. Since almost all the workers are now occupied with running the latest build 20210720 tests.

I gave your VR a bit of a higher prio - it has started and is currently running (for a single test that helps us this much to get this PR moving, this is worth it)

File: lib/x11utils.pm [hunk 1]
```diff
@@ -76,8 +76,16 @@ sub ensure_unlocked_desktop {
...
+        push(@tags, 'blackscreen')      if get_var("DESKTOP") =~ /minimalx|xfce/;    # Only xscreensaver and xfce have a blackscreen as screenlock
+        push(@tags, 'gnome-activities') if check_var('DESKTOP', 'gnome');
         assert_screen \@tags, no_wait => 1;
+        # Starting with GNOME 40, upon login, the activities screen is open (assuming the
+        # user will want to start something. For openQA, we simply press 'esc' to close
+        # it again and really end up on the desktop
+        if (match_has_tag('gnome-activities')) {
+            send_key 'esc';
+            @tags = grep { !/gnome-activities/ } @tags;
+        }
```
Comment (DimStar77) [inline lib/x11utils.pm:88]: Generally ok - just that we have this probably like 3 or 4 times in our codebase already.

We might want to try to further refactor that at one point (can be after we get the tests stable again imho)

**Review by DimStar77 — APPROVED** [review]

**Review by ggardet — APPROVED** [review]

**Review by punkioudi — APPROVED** [review]
Nice!

**Comment by GraceWang571** [comment]
Thank you all for your comments, help and approval.
All the pending tests are almost done. I've updated the VRs parts.

**Pull request merged**